src/main/java/adris/altoclef/tasks/resources/MineAndCollectTask.java

```diff
@@ -194,7 +194,8 @@ protected Vec3d getPos(AltoClef mod, Object obj) {
 
             double blockSq = closestBlock.isEmpty() ? Double.POSITIVE_INFINITY : closestBlock.get().getSquaredDistance(pos, true);
             double dropSq = closestDrop.isEmpty() ? Double.POSITIVE_INFINITY : closestDrop.get().squaredDistanceTo(pos) + 5; // + 5 to make the bot stop mining a bit less
-
+            // TODO: If the drops are close nearby, and picking them up would be faster, do that.
```

Comment:
I understand that we don't want to constantly check how long the fastest path is, but it's annoying and there's probably a good way to fix the overall issue. One time I had Altoclef mining Ancient Debris, and instead of picking up the drop 1 block away, it targeted a really far away block for mining.

Reply:
The main reason this is hard is because you need to pause the user task chain in order to check for dropped items. Communicating between task chains is hard. Perhaps I am misunderstanding. I can answer this more clearly if you want to meet later this week,